framework/helpers/BaseHtml.php

```diff
@@ -1157,6 +1157,18 @@ public static function activeInput($type, $model, $attribute, $options = [])
      */
     public static function activeTextInput($model, $attribute, $options = [])
     {
+        if (!isset($options['maxlength'])) {
```

Comment:
better use array_key_exists to allow explicitly specifying null to not have this attribute rendered.